### ISSUE TITLE
Consent mode (beta)

### ISSUE THREAD

**Issue by devloe** [MatteoGabriele/vue-gtag#393]
Is there a way to send the new consent (beta)?

https://developers.google.com/tag-platform/gtagjs/reference#consent

Using the official library, you can do so with the following code:

gtag('consent', 'default', {
  'ad_storage': 'denied',
  'analytics_storage': 'denied'
});

dataLayer.push({
  'event': 'default_consent'
});

I understand that in vue-gtag there has to be a method called "consent" (just like "set") which at the moment there isn't one.

**Comment by kakauandme** [comment]
@devloe you can use https://matteo-gabriele.gitbook.io/vue-gtag/query for now.

**Comment by MatteoGabriele** [comment]
Indeed, I'd look into the query method. Gtag has a very big api and I just covered the most used once, everything else can easily be accessible via the query method, which is literally just calling the gtag method.

**Comment by EstiT** [comment]
@kakauandme can you provide an example? `query` doesn't have consent as an option, just `config`, `set`, `js` and `event` . Also I'm using `vue-gtag-next` and the page you linked for that version, the code example doesn't seem to work

`const { query } = useGtag();`
`Error: Property 'query' does not exist on type 'VueGtag'.`

**Comment by Nonolanlan1007** [comment]
Hi,
I reopen this issue because I have a TypeScript error when I try to implement the Consent API : 
```ts
query("consent", "update", {
  'ad_storage': value,
  'ad_user_data': value,
  'ad_personalization': value,
  'analytics_storage': value,
})
```

**Comment by croftman** [comment]
> Hi, I reopen this issue because I have a TypeScript error when I try to implement the Consent API :
> 
> query("consent", "update", {
>   'ad_storage': value,
>   'ad_user_data': value,
>   'ad_personalization': value,
>   'analytics_storage': value,
> })

Same here, I open an issue https://github.com/MatteoGabriele/vue-gtag/issues/607

**Comment by MatteoGabriele** [comment]
I am currently rewriting the entire package in TypeScript. I have released a beta version 3.0.0-beta.0, but due to some changes I made, I still need to conduct real-world testing and update the documentation.
I will look into your issue as soon as possible. 
Please let me know if you can check the beta version. It's still an early version, so do not use it in production.
